### PR TITLE
Add password authentification support to SFTP remote storage

### PR DESCRIPTION
**Pull request by ComBin**

### CODE REVIEW

**Comment by AlexAkulov** [comment]
Fix conflicts and tests, please.

File: test/integration/config-sftp.yml [hunk 1]
```diff
@@ -11,8 +11,9 @@ clickhouse:
 sftp:
   address: "sshd"
   username: "root"
-  password: ""
-  key: "/id_rsa"
+  password: "JFzMHfVpvTgEd74XXPq6wARA2Qg3AutJ"
```
Comment (Slach) [inline test/integration/config-sftp.yml]: could you add separatelly config for `config-sftp-auth-key.yaml` and `config-sftp-auth-password.yaml`

File: pkg/new_storage/sftp.go [hunk 2]
```diff
@@ -22,19 +22,24 @@ type SFTP struct {
...
+    if err != nil { return err }
+    sftp_key, err := ssh.ParsePrivateKey(f_sftp_key)
+    if err != nil { return err }
+
+		auth_methods = append(auth_methods, ssh.PublicKeys(sftp_key))
+	}
+
+	if sftp.Config.Password != "" {
+	  auth_methods = append(auth_methods, ssh.Password(sftp.Config.Password))
+	}
```
Comment (Slach) [inline pkg/new_storage/sftp.go:46]: I think we should add 
```
if sftp.Config.Key=="" && sftp.Config.Password=="" {
   return errors.New("please specify sftp.key or sftp.password for authentification")
}
```

i don't think allow users make access to backup without any authorization is a good idea

**Review by Slach — CHANGES_REQUESTED** [review]

File: test/integration/config-sftp.yml (hunk 1 above)
Comment (ComBin) [reply test/integration/config-sftp.yml]: Sure.

File: pkg/new_storage/sftp.go (hunk 2 above)
Comment (ComBin) [reply pkg/new_storage/sftp.go:46]: Without any auth method lib must return some understandable error. But this check make sense too.

**Comment by ComBin** [comment]
About tests. Tests passes on in my local environment, in this time i have no idea why they drops on github. I plans config they in my own repo and try to find root of problem little bit later.

**Comment by Slach** [comment]
> About tests. Tests passes on in my local environment, in this time i have no idea why they drops on github. I plans config they in my own repo and try to find root of problem little bit later.

could you open https://github.com/AlexAkulov/clickhouse-backup/pull/201/checks?check_run_id=2722076505?
```
2021/06/01 19:59:44  info exec clickhouse clickhouse-backup upload test_backup
    integration_test.go:433: 
        	Error Trace:	integration_test.go:433
        	            				integration_test.go:392
        	Error:      	Received unexpected error:
        	            	exit status 2
        	Test:       	TestIntegrationSFTP
```


also, conflicts still present for `sftp.go` please run `git pull upstream master --tags` and merge conflicts

could you use linters for your golang code `snake_case` is not pass for `go fmt` stanard

**Comment by Slach** [comment]
@combin is `sftp` library require `ssh` binary or it's a pure go library?

File: test/integration/config-sftp.yml (hunk 1 above)
Comment (ComBin) [reply test/integration/config-sftp.yml]: Done

File: pkg/new_storage/sftp.go (hunk 2 above)
Comment (ComBin) [reply pkg/new_storage/sftp.go:46]: Done

**Comment by ComBin** [comment]
> could you open https://github.com/AlexAkulov/clickhouse-backup/pull/201/checks?check_run_id=2722076505?

Yes i saw this. But without trace this no helpeble. ((
> also, conflicts still present for sftp.go please run git pull upstream master --tags and merge conflicts
> could you use linters for your golang code snake_case is not pass for go fmt stanard

Variable names replaced. Formatting fixed too.
> is sftp library require ssh binary or it's a pure go library?

SSH binary not needed. Quote from library description:
> Package ssh implements an SSH client and server.

**Comment by ComBin** [comment]
Tests fixed.

**Review by Slach — APPROVED** [review]
thanks a lot for your efforts

**Pull request merged**